File: backend/app/schemas/skill.py
from pydantic import BaseModel, ConfigDict


class SkillResponse(BaseModel):
    id: str
    name: str
    description: str
    short_description: str = ""
    category: str
    author: dict
    version: str
    rating: float = 0.0
    review_count: int = 0
    install_count: int = 0
    price: float = 0.0
    tags: list = []
    capabilities: list = []
    bumblebee_status: str = "pending"
    last_updated: str
    icon: str = ""
    color: str = "#3B82F6"

    model_config = ConfigDict(from_attributes=True)

File: src/types/skill.ts
export interface SkillListing {
  id: string
  name: string
  description: string
  shortDescription: string
  category: 'research' | 'coding' | 'data' | 'security' | 'communication' | 'automation' | 'creative'
  author: {
    name: string
    avatar?: string
    verified: boolean
  }
  version: string
  rating: number
  reviewCount: number
  installCount: number
  price: number
  tags: string[]
  capabilities: string[]
  bumblebeeStatus: 'passed' | 'pending' | 'failed'
  lastUpdated: string
  icon: string
  color: string
}

export interface SkillReview {
  id: string
  skillId: string
  author: string
  rating: number
  text: string
  helpful: number
  timestamp: string
}
